src/igr-tests/igr_functions

```diff
@@ -0,0 +1,231 @@
+        kill "$DINITPID" || ( echo "${TEST_NAME:-}: Cannot stop Dinit instance!" && exit 1 )
+        wait "$DINITPID"
+    fi
+}
+
+# This function spawn a dinit daemon but blocks until dinit exits.
+# It's pass its flags as dinit daemon flags.
+# RESULT: waits for dinit daemon and return exit code of dinit on sccuess.
+#         throw an error on failure.
+spawn_dinit_oneshot() {
+    __find_dinit || error "Cannot find dinit exec path." "Check 'EXEC_PREFIX' variable value."
+    "$DINIT" "$QUIET" -u -d sd -p "$SOCKET" "$@"
+}
+
+# This function find dinitctl and allow access to dinit daemon via dinitctl.
+# It's pass its flags as dinitctl flags.
+# RESULT: Any return code from dinitctl on sccuess.
+#         Throw an error on failure.
+dinitctl() {
+    __find_dinitctl || error "Cannot find dinitctl exec path." "Check 'EXEC_PREFIX' variable value."
+    "$DINITCTL" -p "$SOCKET" "$@"
+}
+
+# This function find dinitcheck and allow access to dinitcheck.
+# It's pass its flags as dinitcheck flags.
+# RESULT: Any return code from dinitcheck on sccuess.
+#         Throw an error on failure.
+dinitcheck() {
+    __find_dinitcheck || error "Cannot find dinitcheck exec path." "Check 'EXEC_PREFIX' variable value."
+    "$DINITCHECK" -d sd "$@"
+}
+
+# This function compare content of given file with expected result.
+# Accpets file as $1 and a text as $2.
+# RESULT: return 0 if content of file is the same with given text.
+#         return 1 if content of file is not the same with given text.
+#         throw an error on any other failure.
+compare_text() {
+    if [ ! -e "$1" ]; then
+        error "$1 file isn't exist!"
+    fi
+    if [ "$(cat "$1" || error "Can't read given file!")" = "$2" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# This function compare content of given file with expected result in another file.
+# Accpets file as $1 and another file as $2.
+# RESULT: return 0 if content of file is the same with another file.
+#         return 1 if content of file is not the same with another file.
+#         throw an error on any other failure.
+compare_file() {
+    if [ ! -e "$1" ]; then
+        error "$1 file isn't exist!"
+    fi
+    if [ ! -e "$2" ]; then
+        error "$2 file isn't exist!"
+    fi
+    if cmp -s "$1" "$2"; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# This function compare command result with content of a file.
+# Accpets file as $1  and expected file as $2.
+# Also if you need to capture stderr, set something as $3 (such as "err").
+# RESULT: set $CMD_OUT variable contain command output.
+#         return 0 if command output is the same with content of file.
+#         return 1 if command output isn't the same with content of file.
+#         throw an error on any other failure.
+compare_cmd() {
+    if [ ! -e "$2" ]; then
+        error "$2 file isn't exist!"
+    fi
+    if [ -n "${3:-}" ]; then
```

Comment:
Make this look for something specific. Eg `if [ "${3:-}" == "err" ]; then`. That gives the possibility to add other options later and also to check if an invalid option is supplied (though you don't have to do that now).